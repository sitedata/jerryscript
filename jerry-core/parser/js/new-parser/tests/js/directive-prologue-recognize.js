/* This test contains directive prologue parser tests. */

"prologue item in quotation mark with semicolon";
"prologue item in quotation mark without semicolon"
'prologue item in aphostrophe with semicolon';
'prologue item in aphostrophe without semicolon'

next_statement;

"string literal in quotation mark with semicolon";
"string literal in quotation mark without semicolon"
'string literal in aphostrophe with semicolon';
'string literal in aphostrophe without semicolon'
