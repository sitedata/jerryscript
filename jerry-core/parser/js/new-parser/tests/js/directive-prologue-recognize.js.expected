
--- Script parsing start ---

  [  1] CBC_PUSH_LITERAL idx:0->string(prologue item in aphostrophe without semicolon)
  [  0] CBC_POP_BLOCK
  [  1] CBC_PUSH_LITERAL idx:1->ident(next_statement)
  [  0] CBC_POP_BLOCK
  [  1] CBC_PUSH_LITERAL idx:2->string(string literal in quotation mark with semicolon)
  [  0] CBC_POP_BLOCK
  [  1] CBC_PUSH_LITERAL idx:3->string(string literal in quotation mark without semicolon)
  [  0] CBC_POP_BLOCK
  [  1] CBC_PUSH_LITERAL idx:4->string(string literal in aphostrophe with semicolon)
  [  0] CBC_POP_BLOCK
  [  1] CBC_PUSH_LITERAL idx:5->string(string literal in aphostrophe without semicolon)
  [  0] CBC_POP_BLOCK

Final byte code dump:

  Maximum stack depth: 1
  Flags: [small_lit_enc]
  Argument range end: 0
  Register range end: 0
  Identifier range end: 1
  Const literal range end: 6
  Literal range end: 6

   0 : CBC_PUSH_LITERAL idx:1(lit)->string(prologue item in aphostrophe without semicolon)
   2 : CBC_POP_BLOCK
   3 : CBC_PUSH_LITERAL idx:0(ident)->ident(next_statement)
   5 : CBC_POP_BLOCK
   6 : CBC_PUSH_LITERAL idx:2(lit)->string(string literal in quotation mark with semicolon)
   8 : CBC_POP_BLOCK
   9 : CBC_PUSH_LITERAL idx:3(lit)->string(string literal in quotation mark without semicolon)
  11 : CBC_POP_BLOCK
  12 : CBC_PUSH_LITERAL idx:4(lit)->string(string literal in aphostrophe with semicolon)
  14 : CBC_POP_BLOCK
  15 : CBC_PUSH_LITERAL idx:5(lit)->string(string literal in aphostrophe without semicolon)
  17 : CBC_POP_BLOCK
  18 : CBC_RETURN_WITH_BLOCK

Byte code size: 19 bytes

Script parsing successfully completed. Total byte code size: 19 bytes

--- Script parsing end ---

